py/ai/h2o/sparkling/ml/features/H2OTargetEncoder.py

```diff
@@ -0,0 +1,107 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.ml.util import JavaMLWritable
+from pyspark.ml.wrapper import JavaEstimator
+from pyspark.sql import SparkSession
+from pyspark import keyword_only
+from py_sparkling.ml.util import getValidatedEnumValue, validateEnumValue
+
+from pysparkling.context import H2OContext
+from pysparkling.spark_specifics import get_input_kwargs
+from ai.h2o.sparkling.ml.params import H2OTargetEncoderParams
+from py_sparkling.ml.util import set_double_values, JavaH2OMLReadable
+from ai.h2o.sparkling.ml.models import H2OTargetEncoderModel
+
+from h2o.utils.typechecks import assert_is_type, Enum
+
+
+class H2OTargetEncoder(H2OTargetEncoderParams, JavaEstimator, JavaH2OMLReadable, JavaMLWritable):
```

Comment:
We should use `JavaMLReadable` instead `JavaMLReadable`. JavaH2OMLReadable was introduced because of the py_sparkling hack.

Also we need a test which serializes & deserializes both unfitted pipeline and model

Reply:
Changed `JavaH2OMLReadable` to `JavaMLReadable` and also added a test covering serde. You can check [here](https://github.com/h2oai/sparkling-water/pull/1354/files#diff-084cb88c118d3617b8cbe7299394efadR70)

Reply:
Thanks! After we switch all algos to ai.h2o, JavaH2OMLReadable is planned to be removed